Include/cpython/code.h

```diff
@@ -26,75 +26,80 @@ typedef uint16_t _Py_CODEUNIT;
+    int co_argcount;              /* #arguments, except *args */               \
+    int co_posonlyargcount;       /* #positional only arguments */             \
+    int co_kwonlyargcount;        /* #keyword only arguments */                \
+    int co_stacksize;             /* #entries needed for evaluation stack */   \
+    int co_firstlineno;           /* first source line number */               \
+                                                                               \
+    /* redundant values (derived from co_localsplusnames and                   \
+       co_localspluskinds) */                                                  \
+    int co_nlocalsplus;           /* number of local + cell + free variables   \
+                                  */                                           \
+    int co_nlocals;               /* number of local variables */              \
+    int co_nplaincellvars;        /* number of non-arg cell variables */       \
+    int co_ncellvars;             /* total number of cell variables */         \
+    int co_nfreevars;             /* number of free variables */               \
+                                                                               \
+    PyObject *co_localsplusnames; /* tuple mapping offsets to names */         \
+    PyObject *co_localspluskinds; /* Bytes mapping to local kinds (one byte    \
+                                     per variable) */                          \
+    PyObject *co_filename;        /* unicode (where it was loaded from) */     \
+    PyObject *co_name;            /* unicode (name, for reference) */          \
+    PyObject *co_qualname;        /* unicode (qualname, for reference) */      \
+    PyObject *co_linetable;       /* bytes (encoding addr<->lineno mapping)    \
+                                     See Objects/lnotab_notes.txt for details. \
+                                  */                                           \
+    PyObject *co_endlinetable;    /* bytes object that holds end lineno for    \
+                                     instructions separated across different   \
+                                     lines */                                  \
+    PyObject *co_columntable;     /* bytes object that holds start/end column  \
+                                     offset each instruction */                \
+                                                                               \
+    PyObject *co_weakreflist;     /* to support weakrefs to code objects */    \
+    /* Scratch space for extra data relating to the code object.               \
+       Type is a void* to keep the format private in codeobject.c to force     \
+       people to go through the proper APIs. */                                \
+    void *co_extra;                                                            \
+    char co_code_adaptive[(SIZE)];                                             \
```

Comment:
Nice!

Reply:
Not a fan of giant macros, myself. They make things harder to parse, both for tools and me.

Maybe something like
```
typedef struct _code_object_fields {
    ... 
} _code_object_fields;

struct PyCodeObject {
    PyObject_VAR_HEAD 
    _code_object_fields body;
    char co_code_adaptive[1];
}
```
Then in the generated code:
```
#define _PyCode_DEF(SIZE) { \
    PyObject_VAR_HEAD  \
    _code_object_fields body; \
    char co_code_adaptive[(SIZE)];    \
}
```
Which avoids most of the code duplication.

Reply:
Ok, let's keep the macro.